core/platform/lf_pico_support.c

```diff
@@ -0,0 +1,329 @@
+#ifdef PLATFORM_PICO
+/*************
+Copyright (c) 2022, The University of California at Berkeley.
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***************/
+
+/**
+ * @brief Raspberry Pi Pico support for the C target of Lingua Franca 
+ * Uses the baremetal pico-sdk 
+ * 
+ * @author{Abhi Gundrala <gundralaa@berkeley.edu>}
+ */
+
+#include "lf_pico_support.h"
+#include "platform.h"
+#include "utils/util.h"
+#include "tag.h"
+
+#include <pico/stdlib.h>
+#include <pico/multicore.h>
+#include <pico/sync.h>
+
+#ifdef LF_UNTHREADED
+// critical section struct binding
+static critical_section_t _lf_crit_sec;
+// semaphore used to notify if sleep was interupted by irq 
+static semaphore_t _lf_sem_irq_event;
+static uint32_t _lf_num_nested_critical_sections = 0;
+#endif
+
+/**
+ * Initialize the LF clock. Must be called before using other clock-related APIs.
+ */
+void lf_initialize_clock(void) {
+    // init stdlib peripherals
+    stdio_init_all();
+    // init sync structs
+    critical_section_init(&_lf_crit_sec);
+    sem_init(&_lf_sem_irq_event, 0, 1);
+    // TODO: use core1
+    multicore_reset_core1();
+}
+
+/**
+ * Fetch the value of an internal (and platform-specific) physical clock and
+ * store it in `t`. in nanoseconds
+ *
+ * Ideally, the underlying platform clock should be monotonic. However, the
+ * core lib tries to enforce monotonicity at higher level APIs (see tag.h).
+ * TODO: might want to use the RTC
```

Comment:
Maybe this should be an ordinary comment in the function body? It refers to the implementation, not the interface.